java-connectors/kafka-connect-azure-eventhubs/src/test/java/io/lenses/streamreactor/connect/azure/eventhubs/source/EventHubsKafkaConsumerControllerTest.java

```diff
@@ -0,0 +1,175 @@
+    List<ConsumerRecord<String, String>> consumerRecordList = Collections.singletonList(consumerRecord);
+
+    ConsumerRecords mockedRecords = mock(ConsumerRecords.class);
+    when(mockedRecords.count()).thenReturn(consumerRecordList.size());
+    when(mockedRecords.iterator()).thenReturn(consumerRecordList.iterator());
+    recordsQueue.put(mockedRecords);
+
+    //when
+    testObj = new EventHubsKafkaConsumerController(mockedBlockingProducer, recordsQueue,
+        inputOutputMap);
+    List<SourceRecord> sourceRecords = testObj.poll(duration);
+
+    //then
+    verify(mockedBlockingProducer).start();
+    verify(mockedDataType, times(2)).getSchema();
+    verify(mockedBlockingProducer, times(2)).getKeyValueTypes();
+    assertNotNull(mockedRecords);
+    assertEquals(1, sourceRecords.size());
+  }
+
+  @Test
+  void pollWithMultipleInputsAndOutputsShouldPollQueueAndReturnSourceRecordsToCorrectOutput()
+      throws InterruptedException {
+    //given
+    Duration duration = Duration.of(2, ChronoUnit.SECONDS);
+    Map<String, String> inputOutputMap = Map.of(INPUT_TOPIC, OUTPUT_TOPIC, INPUT_TOPIC_2,
+        OUTPUT_TOPIC_2);
+
+    SourceDataType mockedDataType = mock(SourceDataType.class);
+    when(mockedDataType.getSchema()).thenReturn(Schema.OPTIONAL_STRING_SCHEMA);
+
+    KeyValueTypes mockedKeyValueTypes = mock(KeyValueTypes.class);
+    when(mockedKeyValueTypes.getKeyType()).thenReturn(mockedDataType);
+    when(mockedKeyValueTypes.getValueType()).thenReturn(mockedDataType);
+
+    KafkaByteBlockingQueuedProducer mockedBlockingProducer = mock(
+        KafkaByteBlockingQueuedProducer.class);
+    when(mockedBlockingProducer.getKeyValueTypes()).thenReturn(mockedKeyValueTypes);
+
+    Headers headersMock = mock(Headers.class);
+    List<Header> emptyHeaderList = Collections.emptyList();
+    when(headersMock.iterator()).thenReturn(emptyHeaderList.iterator());
+
+    ConsumerRecord consumerRecord = mock(ConsumerRecord.class);
+    when(consumerRecord.headers()).thenReturn(headersMock);
+    when(consumerRecord.topic()).thenReturn(INPUT_TOPIC);
+    ConsumerRecord consumerRecord2 = mock(ConsumerRecord.class);
+    when(consumerRecord2.headers()).thenReturn(headersMock);
+    when(consumerRecord2.topic()).thenReturn(INPUT_TOPIC_2);
+    List<ConsumerRecord> consumerRecordList = List.of(consumerRecord, consumerRecord2);
+
+    ConsumerRecords mockedRecords = mock(ConsumerRecords.class);
+    when(mockedRecords.count()).thenReturn(consumerRecordList.size());
+    when(mockedRecords.iterator()).thenReturn(consumerRecordList.iterator());
+    recordsQueue.put(mockedRecords);
+
+    //when
+    testObj = new EventHubsKafkaConsumerController(mockedBlockingProducer, recordsQueue,
+        inputOutputMap);
+    List<SourceRecord> sourceRecords = testObj.poll(duration);
+
+    //then
+    verify(mockedBlockingProducer).start();
+    verify(mockedDataType, times(4)).getSchema();
+    verify(mockedBlockingProducer, times(4)).getKeyValueTypes();
```

Comment:
Why 4?  No context here on why this number.  Please add an assertion hint.

Reply:
There's no assertion so I cannot hint that. I changed this test a bit so it should be more straightforward around that part, hope it'll help!